src/components/landscape-sidebar/species-widget/species-widget-component.jsx

```diff
@@ -0,0 +1,31 @@
+
+import React, { useEffect } from 'react';
+import { loadModules } from '@esri/react-arcgis';
+
+const SpeciesWidgetComponent = ({ map, terrestrialCellData, setSpecies }) => {
+
+ 
+
+  useEffect(() => {
+    if (terrestrialCellData) {
+      loadModules(["esri/layers/FeatureLayer"]).then(([FeatureLayer]) => {
```

Comment:
If we split this logic into two separate functions we will avoid some `fetches` to the API. By loading the layer only on component mount we will fetch layer data just once instead of doing it on every data update. 
This are the metadata `fetches` that are being performed that we would need to do just once:
`https://services9.arcgis.com/IkktFdUAcY3WrH25/arcgis/rest/services/TerrestrialVertebrateSpeciesSHP/FeatureServer?f=json`  
`https://services9.arcgis.com/IkktFdUAcY3WrH25/arcgis/rest/services/TerrestrialVertebrateSpeciesSHP/FeatureServer/0?f=json`

The code could be something along this lines:

```js
const [speciesLayer, setLayer] = useState(null)
 useEffect(() => {
  loadModules(["esri/layers/FeatureLayer"]).then(([FeatureLayer]) => {
    const _speciesLayer = new FeatureLayer({
      // URL to the service
      url: "https://services9.arcgis.com/IkktFdUAcY3WrH25/arcgis/rest/services/TerrestrialVertebrateSpeciesSHP/FeatureServer"
    });
    setLayer(_speciesLayer)
  })
 }, [])

  useEffect(() => {
    if (terrestrialCellData) {
      speciesLayer.definitionExpression = `HBWID IN (${terrestrialCellData.map(i => i.CELL_ID).join(', ')})`;
      const query = speciesLayer.createQuery();
      speciesLayer.queryFeatures(query).then(function(results){
        const { features } = results;
        setSpecies(features.map(c => c.attributes));
      });
    }
  }, [speciesLayer, terrestrialCellData])
```